src/v/cloud_storage/remote_segment.cc

```diff
@@ -315,18 +315,18 @@ remote_segment::offset_data_stream(
     offset_index::find_result pos;
     std::optional<uint16_t> prefetch_override = std::nullopt;
     if (first_timestamp) {
```

Comment:
is it odd that `first_timestamp` wasn't used in this conditional before?

Reply:
`first_timestamp` indicates that this is a time-query. By this point, we have already resolved to the correct segment for the time query. Since we didn't previously used the index, it makes senes that `first_timestamp` wasn't used (we'd just start from the beginning of the segment).